build.assets/tooling/cmd/gen-event-reference/README.md

```diff
@@ -0,0 +1,61 @@
+# Audit event reference generator
+
+The audit event reference generator produces a list of audit events for
+inclusion in the documentation.
+
+## Background
+
+Teleport audit events have unique codes (e.g., `TDB01I`) as well as types (e.g.,
+`user.login`) that are not necessarily unique. Teleport transmits audit events
+as protocol buffer messages that Teleport audit log represents as JSON messages.
+Therefore, to represent audit events, we need to associate codes and types with
+JSON schemas.
+
+The Teleport source assigns audit event codes and types as struct values when
+initializing an audit event. Otherwise, there is nothing intrinsic to an audit
+event that associates its code, type, and schema. Finding all parts of the
+source that emit an audit event is infeasible, so the best we can do is to take
+advantage of the naming conventions we use for declarations of types, codes, and
+schemas.
+
+The reference generator assumes that event codes are constants declared in a
+single file with a declaration name in the following format:
+
+```
+CamelCaseName(Success|Failure)?Code
+```
+
+`CamelCaseName` is a prefix we expect to find across the names of event codes,
```

Comment:
I would add a link to this README in the file where all of the events are defined, and maybe even a Slab page to help people learn these requirements. The audit generator is only useful going forward if we stick to the conventions that you've codified.